File: EY.UbbstractThinkers.ProjectManagementPortal.Server/Models/ApprovalRequest.cs
using System;

namespace EY.UbbstractThinkers.ProjectManagementPortal.Server.Models
{
    public class ApprovalRequest
    {
        public Guid Uid { get; set; }
        public Guid ProjectId { get; set; }
        public Project Project { get; set; }
        public Guid? FromStageId { get; set; }
        public Stage FromStage { get; set; }
        public Guid? ToStageId { get; set; }
        public Stage ToStage { get; set; }
        public Status Status { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime ModifiedAt { get; set; }
        public string ModifiedByUserEmail { get; set; }
    }
}

File: ey.ubbstractthinkers.projectmanagementportal.client/src/models/ApprovalRequest.ts
import type { ApprovalStatus } from "./ApprovalStatus";

export interface ApprovalRequest {
  uid: string;
  projectName: string;
  stageFromName: string;
  stageToName: string;
  status: ApprovalStatus;
  createdAt: Date | null;
  modifiedAt: Date | null;
  modifiedByUserEmail: string;
  createdByUserEmail: string;
}

export const DefaultApprovalRequest: ApprovalRequest = {
  uid: "",
  projectName: "",
  stageFromName: "",
  stageToName: "",
  status: "Default",
  createdAt: null,
  modifiedAt: null,
  modifiedByUserEmail: "",
  createdByUserEmail: "",
};
